python/cudf/cudf/utils/dtypes.py

```diff
@@ -154,6 +154,20 @@ def is_numerical_dtype(obj):
     return dtype.kind in "biuf"
 
 
+def is_integer_dtype(obj):
+    try:
+        dtype = np.dtype(obj)
+    except TypeError:
+        return False
+    return dtype.kind in "iu"
```

Comment:
What about if someone gives us a Pandas nullable integer type?

Reply:
```
>>> pd.Int64Dtype().kind
'i'
```

/shrug

Reply:
Wouldn't it throw at the `np.dtype` call though? `np.dtype(pd.Int64Dtype())`

Reply:
You're right. I made it so that we punt to Pandas if the `np.dtype` conversion fails, instead of always returning `False`.